dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/controller/DrcStubRestController.java

```diff
@@ -29,14 +30,28 @@ public static int getRandomZeroOrOne() {
     }
 
     @PostMapping(value = "/fdc")
-    public Boolean fdc(@NotEmpty @RequestBody final SendContributionFileDataToDrcRequest dataRequest) {
+    public Boolean fdc(@NotNull @RequestBody final SendContributionFileDataToDrcRequest dataRequest) {
         log.info("Request received from DRC to update FDC {}", dataRequest);
         return getRandomZeroOrOne() == 0;
     }
 
     @PostMapping(value = "/contribution")
-    public Boolean contribution(@NotEmpty @RequestBody final SendContributionFileDataToDrcRequest dataRequest) {
+    public Boolean contribution(@NotNull @RequestBody final SendContributionFileDataToDrcRequest dataRequest) {
         log.info("DrcStubRestController Stub returing a response for input {}", dataRequest);
         return getRandomZeroOrOne() == 0;
     }
+
+    @GetMapping(value = "/test")
+    public String testSecure() {
+        log.info("Get Request received to secure test");
+        return "mTLS (secure) Test Get Success";
+    }
+
+    @PostMapping(value = "/test")
+    public String test(@NotNull @RequestBody final SendContributionFileDataToDrcRequest dataRequest) {
+        log.info("Request received from DRC to test FDC {}", dataRequest);
+        return "Request received from DRC to test FDC " + dataRequest;
+    }
```

Comment:
Are these services needed / misnamed?
* The first one isn't really testing mTLS, just connectivity.
* The second one is causing a XSS vulnerability alert in CodeQL (although I think it's a false positive because this service isn't used by a web page).

If we keep them - then could make them more generic / reflect what they actually do:
* `GET /test` that returned 'GET test successful' and
* `POST /test` that returns 'POST test successful: request body length _n_' where _n_ is the length of the POST data in the request body.